clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -938,6 +947,7 @@ public void commitSync(final Map<TopicPartition, OffsetAndMetadata> offsets) {
         acquire();
         try {
             coordinator.commitOffsetsSync(offsets);
+            interceptors.onCommit(offsets);
```

Comment:
Yes, the wrapper around OffsetCommitCallback will still require having the reference to interceptors in AutoCommitTask.
Thanks for pointing out about rebalance and close! I just realized that I missed those cases. 
Agree, I will move interceptor.onCommit call out of KafkaConsumer.commitSync to coordinator.commitOffsetsSync(offsets).